packages/block-editor/src/components/block-navigation/toolbar.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { Button, ToolbarGroup } from '@wordpress/components';
+import { insertAfter, trash } from '@wordpress/icons';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import BlockActions from '../block-actions';
+import BlockMover from '../block-mover';
+
+export default function BlockNavigationToolbar() {
+	const blockClientIds = useSelect( ( select ) =>
+		select( 'core/block-editor' ).getSelectedBlockClientIds()
+	);
```

Comment:
You can optimize this a bit. With the empty dependency array, it will memoize the callback and rerun it on every state change rather than creating a new callback every time.
```suggestion
	const blockClientIds = useSelect( ( select ) => {
		select( 'core/block-editor' ).getSelectedBlockClientIds()
	}, [] );
```

Reply:
Very true. Makes me wonder why this isn't the default for `useSelect`. That ship has sailed now, unfortunately.